### ISSUE TITLE
There does not appear to be a way to customise the built docker images without overriding them

### ISSUE THREAD

**Issue by jameshalsall** [inviqa/harness-base-php#554]
I might have missed something, but in a situation where I want to bundle some additional packages into a built console image, the only option I have of doing this is by overriding the whole `Dockerfile.twig`. Usually, projects then add some kind of `image-dependencies.sh` and do their business in there, with a RUN statement added to the `Dockerfile.twig`.

It would be nice to adopt a siimlar approach to https://github.com/inviqa/harness-go/blob/master/docs/harness-attributes.md#modulesbeforesteps-and-modulesaftersteps if there is not already a way of doing this without overriding the whole Dockerfile that I have missed.

**Comment by andytson-inviqa** [comment]
app build steps cater to this, however they are run as build user, when root user is necessary in places

**Comment by jameshalsall** [comment]
Can you point to the attribute that makes this possible?

**Comment by andytson-inviqa** [comment]
I suppose the other potential pitfalls also is app build is only for the console container (so only files in php-fpm/cron/runner/worker etc is what's then copied from /app to them)
https://github.com/inviqa/harness-base-php/blob/1.1.x/src/_base/docker/image/console/Dockerfile.twig#L56
https://github.com/inviqa/harness-base-php/blob/1.1.x/src/_base/docker/image/console/root/lib/task/build.sh
https://github.com/inviqa/harness-base-php/blob/1.1.x/src/_base/docker/image/console/root/lib/task/build/backend.sh.twig
https://github.com/inviqa/harness-base-php/blob/1.1.x/src/_base/harness/attributes/common.yml#L113-L151

**Comment by jameshalsall** [comment]
That is only executed on a static build as well?

**Comment by andytson-inviqa** [comment]
https://github.com/inviqa/harness-base-php/blob/047f79c2da492792cf1d409f02866b4542769b26/src/_base/harness/scripts/enable.sh.twig#L71

**Comment by jameshalsall** [comment]
Isn't the image already built at that stage? That wouldn't embed things in the image

**Comment by andytson-inviqa** [comment]
dynamic doesn't currently require embedding anything in the image as it's build once, start container, stop container,... start container, so the container and volumes are the state

**Comment by andytson-inviqa** [comment]
however, as I mentioned about the build user, there's definitely a need for a workflow for root user for compiling things and installing system packages, and that could be always in the image

**Comment by jameshalsall** [comment]
If this is by design then great, it's just not clear to me what steps are required to embed something into the built image like I mentioned in the original issue description. The fact that the dynamic/static builds would require two different approaches sounds like duplication to me.

**Comment by andytson-inviqa** [comment]
the different approach is needed as the app build steps require the volume's code, however it's only so different as to where `app build` is run, not what is in it

**Comment by jameshalsall** [comment]
Ok cool, happy to close this then if we already have a solution.

**Comment by andytson-inviqa** [comment]
I think we need this for that pact stuff I tried to port over to this and failed (because build user)

**Comment by jameshalsall** [comment]
Yeah this is the exact scenario I had which made me open this issue, I've currently gone with overriding the Dockerfile.twig for console image and running it from there.
